File: accesa.gamification/src/main/java/ciurezu/gheorghe/dragos/accesa/gamification/data/response/GamificationResponse.java
package ciurezu.gheorghe.dragos.accesa.gamification.data.response;

import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.RequiredArgsConstructor;

@AllArgsConstructor
@RequiredArgsConstructor
@Data
public class GamificationResponse<T> {
    @JsonProperty("payload")
    private T payload;
    @JsonProperty("additional_message")
    private String additionalMessage;
}

File: ClientMobile/app/src/main/java/com/ciurezu/gheorghe/dragos/clientmobile/data/model/GamificationResponse.kt
package com.ciurezu.gheorghe.dragos.clientmobile.data.model

import com.google.gson.annotations.SerializedName

data class GamificationResponse<T>(
    @SerializedName("payload")
    val payload: T,
    @SerializedName("additional_message")
    val additionalMessage: String
)
